tests/components/mqtt/test_discovery.py

```diff
@@ -1273,6 +1372,7 @@ async def test_complex_discovery_topic_prefix(
             }
         },
     )
+    await mqtt_mock_entry_with_yaml_config()
```

Comment:
Is this addition needed?

Reply:
To let the test be representative the MQTT config entry  should be set up. This is done consequently, here it was missed. No error occurs because no manual MQTT items are set up.

Reply:
BTW this is test is not correct at all. It still uses YAML to set up broker parameters, that is deprecated.
And you were right, it is not needed. The entry is pre loaded and async_setup will also trigger setting up the entry.

Reply:
I adjusted the test.